src/engine/controls/cuecontrol.cpp

```diff
@@ -356,6 +356,9 @@ void CueControl::trackLoaded(TrackPointer pNewTrack) {
     CuePosition cuePoint;
     if (pLoadCue) {
         cuePoint.setPosition(pLoadCue->getPosition());
+	// Update load cue source to ensure AnalyzerSilence does not
+	// overwrite it
+	cuePoint.setSource(pLoadCue->getSource());
```

Comment:
The concept of a "source" for cue points is being removed in #2103.

Reply:
As the AnalyzerSilence function shouldUpdateMainCue here: https://github.com/Be-ing/mixxx/blob/f13bb5df5e7c6fac2c76901675b014f141c781ee/src/analyzer/analyzersilence.cpp#L14 no longer uses the cue source, the cue overwriting issue I was experiencing should be solved if #2103 is merged as well. I can revert cuecontrol.cpp if this is the case.

Reply:
I suggest not worrying about it until #2103 is actually merged.